### PR TITLE
fix/modem_driver

### PR DESCRIPTION
**Pull request by AlessioLei94**
- Toggle reset pin when initializing/resetting modem (if present in DTS)
- Enter ready state before releasing script done semaphore
- Add command to write almanac data to modem
- Add commands and Kconfig to set MAGPIO and COEX0 pins usage
- Fix expected response when toggling GNSS

### CODE REVIEW

File: drivers/modem/nordic-nrf9160.h [hunk 1]
```diff
@@ -55,6 +97,8 @@
 #endif
 /* PDP context */
 #define MDM_SETUP_CMD_PDP_CTX "AT+CGDCONT=0,\"IP\",\"" CONFIG_MODEM_NRF9160_APN "\""
+#define MDM_SETUP_CMD_ALMANAC_DATA                                                                 \
+	"AT%XFILEWRITE=1,\"" FACTORY_ALMANAC_DATA_V2 "\",\"" FACTORY_ALMANAC_CHECKSUM_V2 "\""
```
Comment (HesselM) [inline drivers/modem/nordic-nrf9160.h:63]: While i understand (and agree) to have the CMD for the alamanc data in this header, I think it is incorrect to have the almanac actual data in the same header. Interfaces in this header are static. This almanac data is dynamic, it has to be changed over time. 

What if we make a Kconfig of it?
Something like:

```
config MODEM_NORDIC_NRF9160_ALMANAC_DATA
	string "Almanac data"
	default ""
	help
		Custom almanac data

config MODEM_NORDIC_NRF9160_ALMANAC_CHECKSUM
	string "Almanac checksum"
	default ""
	help
		Custom almanac checksum
````

And then in the project-overlay something like:

```
CONFIG_MODEM_NORDIC_NRF9160_ALMANAC_DATA="f0ea020031150900000000000000 ...."
CONFIG_MODEM_NORDIC_NRF9160_ALMANAC_CHECKSUM="dc6e1ee50b4f0cec12f6ee1c039de4926accb054dd74e150db0e622638830b4d"
```

The command would then be something like:

```
#define MDM_SETUP_CMD_ALMANAC_DATA                                                                 \
	"AT%XFILEWRITE=1,\"" CONFIG_MODEM_NORDIC_NRF9160_ALMANAC_DATA "\",\"" CONFIG_MODEM_NORDIC_NRF9160_ALMANAC_CHECKSUM "\""
```

**Review by HesselM — CHANGES_REQUESTED** [review]

File: drivers/modem/nordic-nrf9160.h (hunk 1 above)
Comment (AlessioLei94) [reply drivers/modem/nordic-nrf9160.h:63]: That's a good idea, I'll update it

Comment (AlessioLei94) [reply drivers/modem/nordic-nrf9160.h:63]: Issue with this solution is that Kconfig doesn't allow to have string values on multiple rows, thus the config with the actual alamanac data would be a row with 2000+ columns, not really desirable

Comment (HesselM) [reply drivers/modem/nordic-nrf9160.h:63]: I see. Maybe this approach could works:

- header file in application-folder which holds '#define NRF91_ALMANAC_DATA' and '#define NRF91_ALMANAC_CHECKSUM'
- kconfig of "config MODEM_NORDIC_NRF9160_EXTERNAL_ALMANAC"
- Definition of "#define MDM_SETUP_CMD_ALMANAC_DATA" :
```
#ifdef CONFIG_MODEM_NORDIC_NRF9160_EXTERNAL_ALMANAC
#define MDM_SETUP_CMD_ALMANAC_DATA
    "AT%XFILEWRITE=1,\"" NRF91_ALMANAC_DATA "\",\"" NRF91_ALMANAC_CHECKSUM "\""
#else
#define MDM_SETUP_CMD_ALMANAC_DATA ""
#endif
```

Pre-processing macros/defines are globally scoped and hence the system should be able to pickup the proper data from the header file in the user app. Ofcourse we need to add a proper explanation to the kconfig setting.

Comment (AlessioLei94) [reply drivers/modem/nordic-nrf9160.h:63]: Fixed by adding the header to the application sources and append its content to `autoconf.h` that will be made publicly available from the build system

**Review by HesselM — APPROVED** [review]

**Pull request merged**